contracts/factory/tests/integration.rs

```diff
@@ -308,3 +375,123 @@ fn create_pair() {
     assert_eq!("contract #3", res.contract_addr.to_string());
     assert_eq!("contract #4", res.liquidity_token.to_string());
 }
+
+#[test]
+fn test_pair_migration() {
+    let mut app = mock_app();
+
+    let owner = String::from("owner");
+
+    let token_code_id = store_token_code(&mut app);
+
+    let factory_instance =
+        instantiate_contract(&mut app, &Addr::unchecked(owner.clone()), token_code_id);
+
+    let owner_addr = Addr::unchecked(owner.clone());
+
+    let token_instance0 = instantiate_token(&mut app, token_code_id, &owner_addr, "tokenX");
+    let token_instance1 = instantiate_token(&mut app, token_code_id, &owner_addr, "tokenY");
+    let token_instance2 = instantiate_token(&mut app, token_code_id, &owner_addr, "tokenZ");
+
+    // Create pairs in factory
+    let pairs = [
+        create_pair(
+            &mut app,
+            &owner_addr,
+            &factory_instance,
+            &token_instance0,
+            &token_instance1,
+        ),
+        create_pair(
+            &mut app,
+            &owner_addr,
+            &factory_instance,
+            &token_instance0,
+            &token_instance2,
+        ),
+    ];
+
+    // Change contract ownership
+    let new_owner = Addr::unchecked("new_owner");
+
+    app.execute_contract(
+        owner_addr.clone(),
+        factory_instance.clone(),
+        &ExecuteMsg::ProposeNewOwner {
+            owner: new_owner.to_string(),
+            expires_in: 100,
+        },
+        &[],
+    )
+    .unwrap();
+    app.execute_contract(
+        new_owner.clone(),
+        factory_instance.clone(),
+        &ExecuteMsg::ClaimOwnership {},
+        &[],
+    )
+    .unwrap();
+
+    let pair3 = create_pair(
+        &mut app,
+        &owner_addr,
+        &factory_instance,
+        &token_instance1,
+        &token_instance2,
+    );
+
+    // Should panic due to pairs are not migrated.
+    for pair in pairs.clone() {
+        let res = app
+            .execute_contract(
+                Addr::unchecked("user1"),
+                pair,
+                &PairExecuteMsg::UpdateConfig {
+                    params: Default::default(),
+                },
+                &[],
+            )
+            .unwrap_err();
+
+        assert_eq!(res.to_string(), "Pair is not migrated to new the admin!");
+    }
+
+    // Pair is created after admin migration
+    let res = app
+        .execute_contract(
+            Addr::unchecked("user1"),
+            pair3,
+            &PairExecuteMsg::UpdateConfig {
+                params: Default::default(),
+            },
+            &[],
+        )
+        .unwrap_err();
```

Comment:
Why this should throw an error?

Reply:
It throws [here](https://github.com/astroport-fi/astroport-core/blob/31c8449e6d6e9514910ad0384657304a03d19628/contracts/pair/src/contract.rs#L173). This request is indicative that after mark_as_migrated, the pair is allowed to execute other requests

Reply:
Ok, got it